plasmapy/dispersion/analytical/tests/test_mhd_wave_class.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+import pytest
+
+from astropy import units as u
+
+from plasmapy.dispersion.analytical.mhd_wave_class import mhd_waves
+from plasmapy.formulary.speeds import Alfven_speed, ion_sound_speed
+
+kwargs_wave_limits = {
+    "k": 0.01 * u.rad / u.m,
+    "theta": [0, 90] * u.deg,
+}
+
+
+class TestMHDWave:
+    _kwargs_plasma_cold = {
+        "B": 8.3e-9 * u.T,
+        "ion": "p+",
+        "n_i": 5e6 * u.m**-3,
+    }
+    _kwargs_plasma_hydro = {
+        "B": 0 * u.T,
+        "ion": "p+",
+        "n_i": 5e6 * u.m**-3,
+    }
+    _kwargs_plasma_temp = {
+        "T_e": 1.6e6 * u.K,
+        "T_i": 4.0e5 * u.K,
+    }
+
+    _v_A = Alfven_speed(
+        B=_kwargs_plasma_cold["B"],
+        ion=_kwargs_plasma_cold["ion"],
+        density=_kwargs_plasma_cold["n_i"],
+    )
+    _c_s = ion_sound_speed(
+        **_kwargs_plasma_temp, ion=_kwargs_plasma_cold["ion"], gamma_e=1, gamma_i=3
+    )
+    _c_m = np.sqrt(_v_A**2 + _c_s**2)
+
+    @pytest.mark.parametrize(
+        ("kwargs", "expected"),
+        [
+            (  # Cold plasma limit
+                {**_kwargs_plasma_cold},
+                {
+                    "alfven": [_v_A, 0 * u.m / u.s],
+                    "fast": [_v_A, _v_A],
+                    "slow": [0, 0] * u.m / u.s,
+                },
+            ),
+            (  # No magnetic field
```

Comment:
Good idea to test this limit!

Reply:
Thanks. I actually have a question about this particular limit. Without a magnetic field, the MHD fast mode would reduce to a "sound wave" regardless of the ratio between $T_e$ and $T_i$; however, from the two-fluid model, ion acoustic waves could only exist if $T_e \gg T_i$. Do you think some sort of warning should be added?

Reply:
Good question!  Usually for MHD, it's assumed that the ion & electron temperatures are the same.  Most of the references I've found assume that the sound speed is given by the ideal relation

$$ c_s ≡ \frac{γ k_B T}{m_i} $$

(with $γ = \frac{5}{3}$) which I believe should be used instead of the ion sound speed used for ion acoustic waves. I'll double check this with some friends who work with plasma waves a lot more than I do.  If we assume the same temperature for ions & electrons, we would remove the need for that warning.